libfdcore/peers.c

```diff
@@ -175,7 +183,13 @@ int fd_peer_add ( struct peer_info * info, const char * orig_dbg, void (*cb)(str
 	if (ret) {
 		CHECK_FCT( fd_peer_free(&p) );
 	} else {
-		TRACE_DEBUG(FULL, "Diameter peer %*s added", (int)info->pi_diamidlen, info->pi_diamid);
+		char* peers_str = NULL;
+		size_t peers_str_len;
+		fd_peer_dump_list(&peers_str, &peers_str_len, NULL, 0);
+		TRACE_DEBUG(FULL, "Diameter peer %.*s added", (int)info->pi_diamidlen, info->pi_diamid);
```

Comment:
By the way - http://stackoverflow.com/questions/2239519/is-there-a-way-to-specify-how-many-characters-of-a-string-to-print-out-using-pri says that the `%*s` notation means minimum width (i.e. pad stuff to this length), `%.*s` means maximum width (i.e. truncate at this length).